tests/tracer/test_sampler.py

```diff
@@ -261,6 +261,88 @@ def test_sampling_rule_init():
 
 
+def test_sampling_rule_init_via_env():
+    # Testing single sampling rule
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == "abc"
+        assert len(sampling_rule) == 1
+
+    # Testing multiple sampling rules
+    with override_env(
+        dict(
+            DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz","name":"abc"}, \
+            {"sample_rate":0.5,"service":"my-service","name":"my-name"}]'
+        )
+    ):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == "abc"
+
+        assert sampling_rule[1].sample_rate == 0.5
+        assert sampling_rule[1].service == "my-service"
+        assert sampling_rule[1].name == "my-name"
+        assert len(sampling_rule) == 2
+
+    # Testing for only Sample rate being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == SamplingRule.NO_RULE
+        assert sampling_rule[0].name == SamplingRule.NO_RULE
+        assert len(sampling_rule) == 1
+
+    # Testing for no name being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"service":"xyz"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == "xyz"
+        assert sampling_rule[0].name == SamplingRule.NO_RULE
+        assert len(sampling_rule) == 1
+
+    # Testing for no service being set
+    with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":1.0,"name":"abc"}]')):
+        sampling_rule = DatadogSampler().rules
+        assert sampling_rule[0].sample_rate == 1.0
+        assert sampling_rule[0].service == SamplingRule.NO_RULE
+        assert sampling_rule[0].name == "abc"
+        assert len(sampling_rule) == 1
+
+    # Testing for Sample rate greater than 1.0
+    with pytest.raises(ValueError) as excinfo:
+        with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"sample_rate":2.0,"service":"xyz","name":"abc"}]')):
+            sampling_rule = DatadogSampler().rules
+    assert "SamplingRule(sample_rate={!r}) must be greater than or equal to 0.0 and less than or equal to 1.0" in str(
+        excinfo.value
+    )
+
+    # Testing for no Sample rate
+    with pytest.raises(KeyError) as excinfo:
+        with override_env(dict(DD_TRACE_SAMPLING_RULES='[{"service":"xyz","name":"abc"}]')):
+            sampling_rule = DatadogSampler().rules
+    assert "No sample_rate provided for the following rule" in str(excinfo.value)
```

Comment:
can we assert on the rule being present in the string as well? it should be static/consistent between test runs.

Same comment for the other ones, I'd prefer `"" == str(excinfo.value)` comparisons

Reply:
Same as above